### PR TITLE
Fix openvswitch issues when MicroShift service is disabled

### PR DESCRIPTION
**Pull request by zshi-redhat**
Closes [USHIFT-236](https://issues.redhat.com//browse/USHIFT-236)

### CODE REVIEW

**Comment by ggiguash** [comment]
I think we should also enable openvswitch service in the blueprint:
- https://github.com/openshift/microshift/blob/879bbf1921b68fba5b2b18c4813bd4b485f84674/scripts/image-builder/config/blueprint_v0.0.1.toml#L47
-

**Comment by zshi-redhat** [comment]
> I think we should also enable openvswitch service in the blueprint:
> 
> * https://github.com/openshift/microshift/blob/879bbf1921b68fba5b2b18c4813bd4b485f84674/scripts/image-builder/config/blueprint_v0.0.1.toml#L47

Enabled both openvswitch and NetworkManager in blueprint.

File: hack/cleanup.sh [hunk 1]
```diff
@@ -39,10 +39,11 @@ sudo bash -c '
     echo "Removing /var/lib/microshift"
     crio wipe -f
     systemctl restart crio
+    echo "Reverting microshift-ovs-init.service configuration"
```
Comment (ggiguash) [inline hack/cleanup.sh:42]: Do we need to still pkill the ovn processes above?

File: packaging/rpm/microshift.spec [hunk 2]
```diff
@@ -217,6 +217,7 @@ fi
 sed -i -n -e '/^OPTIONS=/!p' -e '$aOPTIONS="--no-mlockall"' /etc/sysconfig/openvswitch
 %systemd_post microshift-ovs-init.service
 systemctl is-active --quiet NetworkManager && systemctl restart --quiet NetworkManager || true
+systemctl is-active --quiet openvswitch && systemctl restart --quiet openvswitch || true
```
Comment (ggiguash) [inline packaging/rpm/microshift.spec:220]: Should we enable the services here instead of the blueprint?

Comment (ggiguash) [reply packaging/rpm/microshift.spec:220]: We can add the following
`systemctl is-enabled --quiet openvswitch && systemctl enable --quiet openvswitch`

**Review by ggiguash — CHANGES_REQUESTED** [review]

**Comment by oglok** [comment]
Have we tested this and make sure it works? It does not in my local env.

File: hack/cleanup.sh [hunk 3]
```diff
@@ -10,40 +10,47 @@ select yn in "Yes" "No"; do
...
-    systemctl restart crio
-    echo "Reverting microshift-ovs-init.service configuration"
-    /usr/bin/configure-ovs.sh OpenShiftSDN
-    rm -rf /var/lib/microshift
-    rm -rf /var/lib/ovn
-    rm -rf /var/run/ovn
-
-    echo "Cleanup succeeded"
-'
+echo "Stopping microshift"
```
Comment (ggiguash) [inline hack/cleanup.sh:13]: Why did you remove "sudo"? I think we should keep it because it's mentioned without sudo in all the docs, etc.

File: hack/cleanup.sh [hunk 4]
```diff
@@ -10,40 +10,47 @@ select yn in "Yes" "No"; do
...
+set +e
+systemctl stop --now microshift 2>/dev/null
+systemctl disable microshift 2>/dev/null
+systemctl stop --now microshift-containerized 2>/dev/null
+systemctl disable microshift-containerized 2>/dev/null
+podman stop microshift 2>/dev/null
+podman stop microshift-aio 2>/dev/null
+
+echo "Removing crio pods in order"
+NON_OVN_PODS=$(crictl pods | grep -vE ovn | grep -v CREATED | awk '{print $1}');
```
Comment (ggiguash) [inline hack/cleanup.sh:23]: Instead of "grep -v CREATED" -> tail -n +2

File: hack/cleanup.sh [hunk 5]
```diff
@@ -10,40 +10,47 @@ select yn in "Yes" "No"; do
...
+echo "Stopping microshift"
+set +e
+systemctl stop --now microshift 2>/dev/null
+systemctl disable microshift 2>/dev/null
+systemctl stop --now microshift-containerized 2>/dev/null
+systemctl disable microshift-containerized 2>/dev/null
+podman stop microshift 2>/dev/null
+podman stop microshift-aio 2>/dev/null
+
+echo "Removing crio pods in order"
```
Comment (ggiguash) [inline hack/cleanup.sh:22]: What does the comment mean, sorry?

File: hack/cleanup.sh (hunk 4 above)
Comment (ggiguash) [reply hack/cleanup.sh:23]: I think this expression may be bit more selective. I'm a bit concerned with "ovn" pattern.

crictl pods | tail -n +2 | awk '{if ($7~/openshift-ovn-kubernetes/) next; print}' | awk '{print $1}'

**Review by ggiguash — CHANGES_REQUESTED** [review]

File: hack/cleanup.sh [hunk 6]
```diff
@@ -10,40 +10,47 @@ select yn in "Yes" "No"; do
...
+NON_OVN_PODS=$(crictl pods | grep -vE ovn | grep -v CREATED | awk '{print $1}');
+echo $NON_OVN_PODS
```
Comment (oglok) [inline hack/cleanup.sh:24]: Remove this.

File: packaging/rpm/microshift.spec (hunk 2 above)
Comment (zshi-redhat) [reply packaging/rpm/microshift.spec]: Done

**Review by ggiguash — APPROVED** [review]

File: packaging/rpm/microshift.spec [hunk 7]
```diff
@@ -217,6 +217,8 @@ fi
 sed -i -n -e '/^OPTIONS=/!p' -e '$aOPTIONS="--no-mlockall"' /etc/sysconfig/openvswitch
 %systemd_post microshift-ovs-init.service
 systemctl is-active --quiet NetworkManager && systemctl restart --quiet NetworkManager || true
+systemctl is-active --quiet openvswitch && systemctl start --quiet openvswitch || true
+systemctl is-enabled --quiet openvswitch && systemctl enable --quiet openvswitch || true
```
Comment (mangelajo) [inline packaging/rpm/microshift.spec:221]:
```suggestion
systemctl enable --now --quiet openvswitch || true
```

File: packaging/systemd/configure-ovs.sh [hunk 8]
```diff
@@ -4,8 +4,6 @@
 
 set -eux
 
-nmcli conn show br-ex >/dev/null && echo "br-ex already configured" && exit 0
```
Comment (mangelajo) [inline packaging/systemd/configure-ovs.sh:7]: Why are we disabling this now?

Comment (zshi-redhat) [reply packaging/systemd/configure-ovs.sh:7]: I think the rollback or reconfiguration issue was solved by restarting NM after installing NM-ovs.

Comment (zshi-redhat) [reply packaging/systemd/configure-ovs.sh:7]: In the cleanup script, it reverts the init ovs configuration by calling `configure.sh OpenShiftSDN` which doesn't work if we have the br-ex existense check.

Comment (mangelajo) [reply packaging/systemd/configure-ovs.sh:7]: Ahhh ok I missed that detail

**Review by mangelajo — APPROVED** [review]
/lgtm

**Comment by mangelajo** [comment]
Oops I was able to merge it instead of the bot. Do we have a permission issue?
